### PR TITLE
Add Capture and preset load and save metrics

### PR DESCRIPTION
**Pull request by antonrohr**
This belongs together with http://cl/358374603

### CODE REVIEW

File: src/MetricsUploader/MetricsUploader.cpp [hunk 1]
```diff
@@ -6,31 +6,55 @@
...
+  auto timestamp_before = std::chrono::steady_clock::now();
+
+  orbit_base::ImmediateExecutor executer{};
+  return future.Then(&executer, [timestamp_before, log_event_type,
+                                 this](OrbitLogEvent_StatusCode status_code) {
+    auto timestamp_after = std::chrono::steady_clock::now();
+    auto duration =
+        std::chrono::duration_cast<std::chrono::milliseconds>(timestamp_after - timestamp_before);
+    return SendLogEvent(log_event_type, duration, status_code);
+  });
```
Comment (beckerhe) [inline src/MetricsUploader/MetricsUploader.cpp:57]: I'm not sure if a future is the best mechanism here. Usually you would use a RAII here. Something like that:

```cpp
// Sends event with duration when it gets destructed
class ScopedMetric {
    std::chrono::steady_clock::timestamp before_;
    MetricsUploader* uploader_ = nullptr;
 public:
    explicit ScopedMetric(MetricsUploader* uploader) : before_{std::chrono::steady_clock::now()}, uploader_{uploader} {}
    ~ScopedMetric() noexcept {
      if (!uploader_) return;
      auto duration = std::chrono::steady_clock::now() - before_;
      uploader_->SendLogEvent(...);
    }
 };
```

**Review by beckerhe — COMMENTED** [review]
Thanks Anton! I added some questions. :see_no_evil:

File: src/MetricsUploader/Result.cpp [hunk 2]
```diff
@@ -27,6 +27,10 @@ std::string GetErrorMessage(Result result) {
       return "Connection to metrics uploader wasn't closed";
     case kNoError:
       return "No errors";
+    case kUnknownStatusCode:
+      return "Unknown status code";
```
Comment (IrinaShkviro) [inline src/MetricsUploader/Result.cpp:31]: I'm not sure that it's very clear what was the error here. Maybe we should add a bit more details here (specify that the OrbitLogEvent::StatusCode in library doesn't have the code)?

File: src/OrbitGl/App.cpp [hunk 3]
```diff
@@ -741,6 +769,12 @@ PresetLoadState OrbitApp::GetPresetLoadState(
 }
 
 ErrorMessageOr<void> OrbitApp::OnSaveCapture(const std::string& file_name) {
+  orbit_base::Promise<orbit_metrics_uploader::OrbitLogEvent_StatusCode> promise;
+  if (metrics_uploader_ != nullptr) {
+    (void)metrics_uploader_->SendTimedStatusEvent(
+        orbit_metrics_uploader::OrbitLogEvent_LogEventType_ORBIT_PRESET_SAVE, promise.GetFuture());
```
Comment (IrinaShkviro) [inline src/OrbitGl/App.cpp:775]:
```suggestion
         orbit_metrics_uploader::OrbitLogEvent_LogEventType_ORBIT_CAPTURE_SAVE, promise.GetFuture());
```

**Comment by antonrohr** [comment]
I will close this PR soon and open another one with `ScopedMetric`. I leave it open for now, so I don't forget to address the comments in the new PR

**Pull request closed without merging**